internal/gengapic/metadata.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gengapic
+
+import (
+	"google.golang.org/genproto/googleapis/gapic/metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func (g *generator) genGapicMetadataFile() error {
+	data, err := protojson.MarshalOptions{Multiline: true}.Marshal(g.metadata)
```

Comment:
This is fine for now, but if it becomes an issue down the road we may want to stabilize this output to reduce diffs.

Reply:
I _think_ this writes maps by the keys in lexicographical order. But agreed, the public docs state not to depend on stable output.  Let's keep an eye on it as you've suggested.